### PR TITLE
New Navigation Bar

### PR DESCRIPTION
**Pull request by mrMarshalX**
# What

- implemented new component to render navigation bar on the left hand side of the screen

## Testing

- [x] Is this change covered by the unit tests?

## Compatibility

- [x] Does this change maintain backward compatibility?

## Screenshots
image
image
image
image

### CODE REVIEW

File: src/components/navigationBar/NavigationBar.spec.tsx [hunk 1]
```diff
@@ -0,0 +1,63 @@
...
+
+  const hamburger = container.querySelector("svg");
+
+  if (hamburger) {
+    userEvent.click(hamburger);
+  }
+
+  if (container.firstChild) {
+    fireEvent.transitionEnd(container.firstChild);
+  }
```
Comment (nowyDEV) [inline src/components/navigationBar/NavigationBar.spec.tsx:40]: I think it's better to make type assertion in these cases.
If something brakes, your if statement won't fire and the tests might pass.
With assertion it should throw an error.

File: src/components/navigationBar/types.ts [hunk 2]
```diff
@@ -0,0 +1,40 @@
...
+  caption: string;
+  icon: PathName;
+  children?: React.ReactNode;
+};
+
+export type MenuItemProps = Item & {
+  expanded: boolean;
+  expandedDone: boolean;
+  popoverConfig?: PopoverConfig;
+};
```
Comment (nowyDEV) [inline src/components/navigationBar/types.ts:30]:
```suggestion
export type MenuItemProps = {
  caption: string;
  icon: PathName;
  children?: React.ReactNode;
};

export type TriggerProps = {
  expanded: boolean;
  expandedDone: boolean;
  popoverConfig?: PopoverConfig;
};
```

File: src/components/navigationBar/MenuItem.tsx [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+import { Flex } from "../container";
+import { ExtendedPopover } from "../extendedPopover";
+import { ExtendedTooltip } from "../extendedTooltip";
+import { SVGIcon } from "../svgicon";
+import { Trigger } from "./components/Trigger";
+import { StyledTriggerWrapper } from "./styles";
+import { Item } from "./types";
+import { useMenuBarConfiguration } from "./utils/hooks";
+
+export function MenuItem({ caption, icon, children }: Item) {
```
Comment (nowyDEV) [inline src/components/navigationBar/MenuItem.tsx:12]: Some misalignment with the naming of types?

```suggestion
export function MenuItem({ caption, icon, children }: MenuItemProps) {
```

File: src/components/navigationBar/NavigationBar.tsx [hunk 4]
```diff
@@ -0,0 +1,63 @@
...
+    duration: "0.2s",
+  },
+  children,
+}: Props) {
+  const { expanded, expandedDone, handleExpand, setExpanded, setExpandedDone } =
+    useMenuBarConfiguration();
+
+  if (children == null) {
+    return null;
+  }
```
Comment (nowyDEV) [inline src/components/navigationBar/NavigationBar.tsx:34]: Maybe we could add console.warn here to notify the user that he passed `null` and component won't work?

File: src/components/navigationBar/styles.ts [hunk 5]
```diff
@@ -0,0 +1,102 @@
...
+  position: ${({ position }) => position};
+  top: 0;
+  bottom: 0;
+  left: 0;
+  width: ${({ expanded }) => (expanded ? "14rem" : theme.spacing.spacing07)};
+  background-color: ${({ backgroundColor }) => backgroundColor};
+  padding: ${theme.spacing.spacing05}
+    ${({ expanded }) =>
+      expanded ? theme.spacing.spacing04 : theme.spacing.spacing03}
+    ${theme.spacing.spacing04};
```
Comment (nowyDEV) [inline src/components/navigationBar/styles.ts:44]: Maybe it's better to be explicit here (padding-left/right/top/bottom), it could improve readability.

**Comment by nowyDEV** [comment]
Let's remove those {""}, they were probably added by prettier
Screenshot_2022-02-14_11-40-34

**Comment by nowyDEV** [comment]
Can we hide vertical scrollbars?
Screenshot_2022-02-14_11-41-38

**Comment by nowyDEV** [comment]
Scrollable shrinks elements because it adds sidebar that takes up place.
Screenshot_2022-02-14_11-43-35

File: src/components/navigationBar/NavigationBar.spec.tsx (hunk 1 above)
Comment (mrMarshalX) [reply src/components/navigationBar/NavigationBar.spec.tsx:40]: We still need those `ifs`. Without it `userEvent.click` throws type error as `hamburger` can be `null`.

image

Nevertheless, I have added assertions for more readability in test report.

File: src/components/navigationBar/styles.ts (hunk 5 above)
Comment (mrMarshalX) [reply src/components/navigationBar/styles.ts:44]: 👍

**Review by nowyDEV — COMMENTED** [review]
forgot to remove [nordcloud-gnui-4.14.0.tgz](https://github.com/nordcloud/GNUI/pull/429/files#diff-648a1e852790f2fcb978fa4392ed791f7216977578bf8b68f3c8a6953f9da200)

File: src/components/navigationBar/NavigationBar.spec.tsx (hunk 1 above)
Comment (nowyDEV) [reply src/components/navigationBar/NavigationBar.spec.tsx]: Type assertion -> `userEvent.click(hamburger as SVGSVGElement)`
that's what i meant, this should remove the error

File: package.json [hunk 6]
```diff
@@ -1,7 +1,7 @@
 {
   "name": "@nordcloud/gnui",
   "description": "Nordcloud Design System - a collection of reusable React components used in Nordcloud's SaaS products",
-  "version": "4.13.0",
+  "version": "4.14.0",
```
Comment (Bosmanfrx) [inline package.json:4]: I think we need to adjust this version now ;)

**Review by nowyDEV — APPROVED** [review]

**Review by nowyDEV — APPROVED** [review]

**Review by Bosmanfrx — APPROVED** [review]

**Review by nowyDEV — APPROVED** [review]

**Pull request merged**